### PR TITLE
ncm-metaconfig: xinetd: add log_on_success and log_on_failure

### PR DESCRIPTION
**Pull request by gombasg**
Add two more xinetd options.

Unfortunately, with metaconfig, it is not really possible to add new options from templates only, as the `.tt` file shipped by the RPM needs to know the type of the new options. It would be nice if Template-Toolkit would be able to infer the type from the profile...

### CODE REVIEW

**Comment by jrha** [comment]
You can check the types of arbitary values in TT with CCM:
```
[%-             IF CCM.is_list(pair.value) -%]
 [%                 pair.value.join(' ') %]
```

**Comment by gombasg** [comment]
> You can check the types of arbitary values in TT with CCM:
> 
> ```
> [%-             IF CCM.is_list(pair.value) -%]
>  [%                 pair.value.join(' ') %]
> ```

Oh, nice :-) Is there also a way to query the type in a form which would be suitable for a `CASE` statement instead of a series of `IF ... ELSE IF ... ELSE IF ...`?

**Comment by jrha** [comment]
> > You can check the types of arbitary values in TT with CCM:
> > ```
> > [%-             IF CCM.is_list(pair.value) -%]
> >  [%                 pair.value.join(' ') %]
> > ```
> 
> Oh, nice :-) Is there also a way to query the type in a form which would be suitable for a `CASE` statement instead of a series of `IF ... ELSE IF ... ELSE IF ...`?

Uhm... pass.

**Comment by stdweird** [comment]
@gombasg how would you want to see this? we can add anything we want, it's as simple as eg adding a new entry in https://github.com/quattor/CCM/blob/master/src/main/perl/TextRender.pm#L473

**Comment by stdweird** [comment]
jenkins test this please

**Comment by jrha** [comment]
I've opened an issue with the request for TextRender enhancement.

**Comment by jrha** [comment]
```
2022-10-07T10:40:11.4897914Z # Looks like you failed 16 tests of 122.
2022-10-07T10:40:11.4978193Z src/test/perl/service-xinetd.t ........... 
…
2022-10-07T10:40:11.4999754Z not ok 43 - Found at least one match (total 0 matches) for test idx 2 (pattern (?^:(?m:^\s{4}disable\s=\sno$)))
2022-10-07T10:40:11.5000241Z not ok 44 - Order ok for test idx 2 (pattern (?^:(?m:^\s{4}disable\s=\sno$))) (lastpos 14 before -1)
2022-10-07T10:40:11.5000768Z not ok 45 - Found at least one match (total 0 matches) for test idx 3 (pattern (?^:(?m:^\s{4}log_on_failure\s=\sHOST USERID$)))
2022-10-07T10:40:11.5001674Z not ok 46 - Order ok for test idx 3 (pattern (?^:(?m:^\s{4}log_on_failure\s=\sHOST USERID$))) (lastpos 14 before -1)
2022-10-07T10:40:11.5002254Z not ok 47 - Found at least one match (total 0 matches) for test idx 4 (pattern (?^:(?m:^\s{4}log_on_success\s=\sHOST DURATION TRAFFIC$)))
2022-10-07T10:40:11.5002822Z not ok 48 - Order ok for test idx 4 (pattern (?^:(?m:^\s{4}log_on_success\s=\sHOST DURATION TRAFFIC$))) (lastpos 14 before -1)
…
2022-10-07T10:40:11.5007400Z not ok 57 - Found at least one match (total 0 matches) for test idx 9 (pattern (?^:(?m:^\s{4}wait\s=\sno$)))
2022-10-07T10:40:11.5007874Z not ok 58 - Order ok for test idx 9 (pattern (?^:(?m:^\s{4}wait\s=\sno$))) (lastpos 112 before -1)
…
2022-10-07T10:40:11.5018563Z not ok 79 - Found at least one match (total 0 matches) for test idx 2 (pattern (?^:(?m:^\s{4}disable\s=\sno$)))
2022-10-07T10:40:11.5019063Z not ok 80 - Order ok for test idx 2 (pattern (?^:(?m:^\s{4}disable\s=\sno$))) (lastpos 13 before -1)
…
2022-10-07T10:40:11.5024528Z not ok 91 - Found at least one match (total 0 matches) for test idx 8 (pattern (?^:(?m:^\s{4}wait\s=\syes$)))
2022-10-07T10:40:11.5025052Z not ok 92 - Order ok for test idx 8 (pattern (?^:(?m:^\s{4}wait\s=\syes$))) (lastpos 123 before -1)
…
2022-10-07T10:40:11.5035818Z not ok 113 - Found at least one match (total 0 matches) for test idx 2 (pattern (?^:(?m:^\s{4}disable\s=\sno$)))
2022-10-07T10:40:11.5036306Z not ok 114 - Order ok for test idx 2 (pattern (?^:(?m:^\s{4}disable\s=\sno$))) (lastpos 13 before -1)
…
2022-10-07T10:40:11.5038772Z not ok 119 - Found at least one match (total 0 matches) for test idx 5 (pattern (?^:(?m:^\s{4}wait\s=\sno$)))
2022-10-07T10:40:11.5039254Z not ok 120 - Order ok for test idx 5 (pattern (?^:(?m:^\s{4}wait\s=\sno$))) (lastpos 97 before -1)
```

**Comment by jrha** [comment]
Ah, the booleans are being rendered as integers:
```
service time
{
    disable = 0
    id = time-stream
    socket_type = stream
    type = INTERNAL
    user = root
    wait = 0
}
```

File: ncm-metaconfig/src/main/metaconfig/xinetd/main.tt [hunk 1]
```diff
@@ -1,16 +1,13 @@
-[%- booleans = ["disable", "wait"] -%]
-[%- lists = ["type", "flags", "only_from", "cps"] -%]
 service [% servicename %]
 {
 [% FILTER indent -%]
 [%-     FOREACH pair IN options.pairs -%]
 [%          pair.key %] =
-[%-         SWITCH pair.key -%]
-[%-             CASE booleans -%]
+[%-         IF CCM.is_boolean(pair.value) -%]
```
Comment (jrha) [inline ncm-metaconfig/src/main/metaconfig/xinetd/main.tt:6]:
```suggestion
[%-         IF pair.value.is_boolean -%]
```

**Comment by jrha** [comment]
Down to only four test failures now:
```
2022-10-07T11:17:37.9564035Z not ok 45 - Found at least one match (total 0 matches) for test idx 3 (pattern (?^:(?m:^\s{4}log_on_failure\s=\sHOST USERID$)))
2022-10-07T11:17:37.9564716Z not ok 46 - Order ok for test idx 3 (pattern (?^:(?m:^\s{4}log_on_failure\s=\sHOST USERID$))) (lastpos 16 before -1)
2022-10-07T11:17:37.9565448Z not ok 47 - Found at least one match (total 0 matches) for test idx 4 (pattern (?^:(?m:^\s{4}log_on_success\s=\sHOST DURATION TRAFFIC$)))
2022-10-07T11:17:37.9566176Z not ok 48 - Order ok for test idx 4 (pattern (?^:(?m:^\s{4}log_on_success\s=\sHOST DURATION TRAFFIC$))) (lastpos 16 before -1)
```

File: ncm-metaconfig/src/main/metaconfig/xinetd/tests/profiles/rsync.pan [hunk 2]
```diff
@@ -2,3 +2,7 @@ object template rsync;
 
 include 'metaconfig/xinetd/services/rsync';
 
+prefix "/software/components/metaconfig/services/{/etc/xinetd.conf/rsync}/contents";
```
Comment (jrha) [inline ncm-metaconfig/src/main/metaconfig/xinetd/tests/profiles/rsync.pan:5]:
```suggestion
prefix "/software/components/metaconfig/services/{/etc/xinetd.d/rsync}/contents";
```

**Review by jrha — APPROVED** [review]
Now fixed and LGTM.

**Review by ned21 — APPROVED** [review]

**Pull request merged**